### PR TITLE
Autocomplete: clean up settings

### PR DESCRIPTION
**Pull request by valerybugakov**
TLDR: clean up autocomplete settings and improve visibility/debuggability of autocomplete provider creation errors.

- Changed the exposed values for `cody.autocomplete.advanced.provider` to:
    - `default` — our recommended setup.
    - `experimental-ollama` — for Ollama users.
- Added proper descriptions to `cody.autocomplete.advanced.provider` and its options.
    - Screenshot 2024-09-16 at 12 02 23
- Hidden `cody.autocomplete.advanced.model`. It's kept for internal testing purposes only.
- Removed unused `cody.autocomplete.advanced.serverEndpoint` and `cody.autocomplete.advanced.accessToken`.
- Replaced provider value validation based on enums specified in package.json with the runtime error surfaced to the user at the end of the autocomplete provider creation calls. The same error is logged to the output channel for debugging purposes. The error is different depending on the provider ID source and instance type:
    - Screenshot 2024-09-16 at 11 56 54
    - Screenshot 2024-09-16 at 11 56 28
    - Screenshot 2024-09-16 at 11 54 17
- Closes https://linear.app/sourcegraph/issue/CODY-3744/revise-autocomplete-provider-settings-based-on-user-feedback
- [Slack thread](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1725858507331409?thread_ts=1725637232.112709&cid=C04MSD3DP5L) with related discussion.

## Test plan

1. Search for the `cody.autocomplete.advanced.provider` setting in VS Code settings UI and ensure it matches changes from this PR description (updated description and options).
2. Add "cody.autocomplete.advanced.provider": "random-value" to your settings.json file and observe the error autocomplete error message in the standard VS Code error tooltip.
3. CI with updated unit tests.

## Changelog

- Autocomplete: removed the `cody.autocomplete.advanced.model` setting and updated supported values for `cody.autocomplete.advanced.provider`.

### CODE REVIEW

**Review by sqs — APPROVED** [review]

File: vscode/package.json [hunk 1]
```diff
@@ -1115,31 +1115,13 @@
...
-            "fireworks",
-            "unstable-gemini",
-            "unstable-openai",
-            "experimental-ollama",
-            "experimental-openaicompatible"
+          "default": "default",
+          "enum": ["default", "experimental-ollama"],
+          "enumDescriptions": [
+            "Our recommended setup with the best balance of quality and latency. We continuously update this for optimal performance.",
+            "Experimental support for Ollama users. Use `cody.autocomplete.experimental.ollamaOptions` to configure requests to Ollama server."
```
Comment (valerybugakov) [inline vscode/package.json:1122]: @aramaraju for proofreading and suggesting better wording if needed.

**Comment by aramaraju** [comment]
What happens to users who have a different setting at an earlier version? Do we do an one-time migration when they upgrade to `default`?

**Comment by valerybugakov** [comment]
> What happens to users who have a different setting at an earlier version?

Their setting is still valid because it's saved in the settings.json file. But they won't be able to select this value from the UI anymore.
